apps/mobile/src/libs/components/gradient-slider/gradient-slider.tsx

```diff
@@ -0,0 +1,115 @@
+import Slider from "@react-native-community/slider";
+import React from "react";
+import { type LayoutChangeEvent } from "react-native";
+
+import {
+	Icon,
+	LinearGradient,
+	Planet,
+	Text,
+	View,
+} from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/enums";
+import {
+	generateGradientColors,
+	generateGradientLocations,
+} from "~/libs/helpers/helpers";
+import { useCallback, useState } from "~/libs/hooks/hooks";
+import { type colorToGradientColors } from "~/libs/maps/maps";
+import { globalStyles } from "~/libs/styles/styles";
+
+import { styles } from "./styles";
+
+type Properties = {
+	gradientColors: keyof typeof colorToGradientColors;
+	max: number;
+	min: number;
+};
+
+const INITIAL_SLIDER_WIDTH = 0;
+const MAX_PERCENT = 100;
+const PLANET_TRANSLATE_OFFSET = 3;
+const ICON_TRANSLATE_OFFSET = 20;
+
+const GradientSlider: React.FC<Properties> = ({ gradientColors, max, min }) => {
```

Comment:
I think we should also pass an `onValueChange` callback and call it inside `handleValueChange` without keeping all the value-changing logic inside the Slider itself. For now, you can make it optional and we'll add the logic later.